Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift

```diff
@@ -50,13 +53,65 @@ final class GenerativeContentRemoteTests: XCTestCase {
         }
     }
 
+    func test_generateText_with_failure_returns_error_when_token_generation_fails() async throws {
+        // Given
+        let remote = GenerativeContentRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: "sites/\(sampleSiteID)/jetpack-openai-query/jwt", filename: "jwt-token-failure")
+        network.simulateResponse(requestUrlSuffix: "text-completion", filename: "generative-text-failure")
+
+        // When
+        await assertThrowsError {
+            _ = try await remote.generateText(siteID: sampleSiteID,
+                                              base: "generate a product description for wapuu pencil",
+                                              feature: .productDescription)
+        } errorAssert: { error in
+            // Then
+            error as? WordPressApiError == .unknown(code: "oauth2_invalid_token", message: "The OAuth2 token is invalid.")
+        }
+    }
+
+    func test_generateText_retries_after_regenarating_token_upon_receiving_403_error() async throws {
+        // Given
+        let jwtRequestPath = "sites/\(sampleSiteID)/jetpack-openai-query/jwt"
+        let textCompletionPath = "text-completion"
+        let remote = GenerativeContentRemote(network: network)
+        network.simulateResponse(requestUrlSuffix: jwtRequestPath, filename: "jwt-token-success")
+        network.simulateResponse(requestUrlSuffix: textCompletionPath, filename: "generative-text-success")
+
+        // When
+        _ = try await remote.generateText(siteID: sampleSiteID,
+                                          base: "generate a product description for wapuu pencil",
+                                          feature: .productDescription)
+        // Then
+        XCTAssertEqual(numberOfJwtRequests(in: network.requestsForResponseData), 1)
+
+        // When
+        _ = try await remote.generateText(siteID: sampleSiteID,
+                                          base: "generate a product description for wapuu pencil",
+                                          feature: .productDescription)
+
+        // Then
+        XCTAssertEqual(numberOfJwtRequests(in: network.requestsForResponseData), 1)
```

Comment:
❓ Is this redundant? If not, could you add a comment explaining why we must repeat this step?

Reply:
This is checking that JWT is not requested again until "invalid token error" is received. Added comment in https://github.com/woocommerce/woocommerce-ios/pull/10218/commits/1a1561278eae0f2e4e40e70dbe66f1359b347f08